cpp/src/search/search.cu

```diff
@@ -129,4 +130,53 @@ gdf_column upper_bound(table const& t,
   return detail::search_ordered(t, values, false, desc_flags, nulls_as_largest);
 }
 
+template <bool nullable = true>
+struct compare_with_value{
+
+            compare_with_value(device_table t, device_table val, bool nulls_are_equal = true)
+                : compare(t, val, nulls_are_equal) {}
+
+            __device__ bool operator()(gdf_index_type i){
+               return compare(i, 0);
+            }
+            row_equality_comparator<nullable> compare;
+};
+
+bool contains(gdf_column const& column, gdf_scalar const& value)
+{
+  // No element to compare against
+  if (column.size == 0 || value.is_valid == false) {
+      return false;
+  }
+
+  cudaStream_t stream = 0;
```

Comment:
Create an API in the `detail` namespace that accepts a stream argument defaulted to `0` and make the public API simply call the `detail::contains` function.